File: problems/hard/DNA_health/__tests__/test_dna_health.js
const { computeDnaHealth } = require('./dna_health');

function runSampleTest() {
    const genes = ['a', 'b', 'c', 'aa', 'd', 'b'];
    const health = [1, 2, 3, 4, 5, 6];
    const strands = [
        { start: 1, end: 5, d: 'caaab' },  // health = 19
        { start: 0, end: 4, d: 'xyz' },    // health = 0
        { start: 2, end: 4, d: 'bcdybc' }  // health = 6
    ];

    const result = computeDnaHealth(genes, health, strands);
    console.log('Expected: 0 19');
    console.log(`Got: ${result.min} ${result.max}`);

    if (result.min === 0 && result.max === 19) {
        console.log('✅ Test passed!');
    } else {
        console.error('❌ Test failed.');
    }
}

runSampleTest();

File: problems/hard/DNA_health/__tests__/test_dna_health.c
#include <stdio.h>
#include <string.h>

void computeDnaHealth(int n, char* genes[], int health[], int strands[][3], char* strandStrings[], int numStrands, int* minHealth, int* maxHealth);

void test_dna_health() {
    int n = 6;
    char* genes[] = {"a", "b", "c", "aa", "d", "b"};
    int health[] = {1, 2, 3, 4, 5, 6};

    int strands[3][3] = {{1, 5}, {0, 4}, {2, 4}};
    char* strandStrings[] = {"caaab", "xyz", "bcdybc"};
    int minHealth, maxHealth;

    computeDnaHealth(n, genes, health, strands, strandStrings, 3, &minHealth, &maxHealth);

    if (minHealth != 0 || maxHealth != 19) {
        printf("❌ Test failed: Expected 0 19, got %d %d\n", minHealth, maxHealth);
    } else {
        printf("✅ All C tests passed!\n");
    }
}

int main() {
    test_dna_health();
    return 0;
}
